src/stash/src/lib.rs

```diff
@@ -702,6 +702,63 @@ where
             })
             .await
     }
+
+    /// Transactionally deletes any kv pair from `self` which returns `true` for
+    /// `predicate`.
+    ///
+    /// Note that this operation:
+    /// - Runs in a single transaction and cannot be combined with other
+    ///   transactions.
+    /// - Scans the entire table to perform deletions.
+    #[tracing::instrument(level = "debug", skip_all)]
+    pub async fn delete<P>(&self, stash: &mut Stash, predicate: P) -> Result<(), StashError>
+    where
+        P: Fn(&K, &V) -> bool,
+        K: Hash + Clone,
+        V: Clone,
+    {
+        let initial = self.peek_one(stash).await?;
```

Comment:
These now use the old style API. After this merges I'll change them to use the new txn API.

Reply:
This code was ancient; hadn't realize the API changed. Believe I migrated it to the new API but PTAL and lmk if I've done that correctly.

Reply:
Going to merge and then can fix in post if need be.